planner/core/rule_predicate_push_down.go

```diff
@@ -118,6 +127,7 @@ func (p *LogicalUnionScan) PredicatePushDown(predicates []expression.Expression)
 func (ds *DataSource) PredicatePushDown(predicates []expression.Expression) ([]expression.Expression, LogicalPlan) {
 	predicates = expression.PropagateConstant(ds.ctx, predicates)
 	predicates = DeleteTrueExprs(ds, predicates)
+	predicates = ds.AddPrefix4ShardIndexes(ds.ctx, predicates)
```

Comment:
Should we do the transformation processing push donw conditions or doing the expression rewrite or generation(https://github.com/pingcap/tidb/blob/master/planner/core/expression_rewriter.go#L1031)?
I'm not sure if the expression index filter range could be calculated automatically by now.